agent/consul/wanfed/wanfed.go

```diff
@@ -136,42 +131,24 @@ func (t *Transport) DialAddressTimeout(addr memberlist.Address, timeout time.Dur
 	}
 
 	if dc != t.datacenter {
-		gwAddr := t.gwResolver(dc)
-		if gwAddr == "" {
-			return nil, structs.ErrDCNotAvailable
-		}
-
-		return t.dial(dc, node, pool.ALPN_WANGossipStream, gwAddr)
+		return t.dial(dc, node, pool.ALPN_WANGossipStream)
 	}
 
 	return t.IngestionAwareTransport.DialAddressTimeout(addr, timeout)
 }
 
-// NOTE: There is a close mirror of this method in agent/pool/pool.go:DialTimeoutWithRPCType
-func (t *Transport) dial(dc, nodeName, nextProto, addr string) (net.Conn, error) {
-	wrapper := t.tlsConfigurator.OutgoingALPNRPCWrapper()
-	if wrapper == nil {
-		return nil, fmt.Errorf("wanfed: cannot dial via a mesh gateway when outgoing TLS is disabled")
-	}
-
-	dialer := &net.Dialer{Timeout: 10 * time.Second}
-
-	rawConn, err := dialer.Dial("tcp", addr)
-	if err != nil {
-		return nil, err
-	}
-
-	if tcp, ok := rawConn.(*net.TCPConn); ok {
-		_ = tcp.SetKeepAlive(true)
-		_ = tcp.SetNoDelay(true)
-	}
-
-	tlsConn, err := wrapper(dc, nodeName, nextProto, rawConn)
-	if err != nil {
-		return nil, err
-	}
-
-	return tlsConn, nil
+func (t *Transport) dial(dc, nodeName, nextProto string) (net.Conn, error) {
```

Comment:
Instead of ending up with 3 similar copies of the same function structure I refactored stuff so that now there's only one implementation of this dialing approach shared by all 3 locations.